### PR TITLE
[BTS-346] Popover

### PR DESCRIPTION
**Pull request by rapahaeru**
Implements theme in the Popover component
https://jirasoftware.catho.com.br/browse/BTS-346

### CODE REVIEW

File: components/Popover/arrowProperties.js [hunk 1]
```diff
@@ -4,16 +4,26 @@ const ARROW_SIZE = '16px';
...
+    components: {
+      popover: { skins },
+    },
+  },
+  inverted,
+) => {
+  if (inverted) {
+    return skins[skin].text;
+  }
+  return skins[skin].background;
```
Comment (ddsilva) [inline components/Popover/arrowProperties.js]: Suggestion:

```js
const getColorBySkin = (
  skin,
  inverted,
  {
    components: {
      popover: { skins },
    },
  },
) => inverted ? skins[skin].text : skins[skin].background;
```

**Comment by ddsilva** [comment]
The theme isn't passed down to `Content` component, from `Popover`.

File: components/Popover/sub-components/Content.jsx [hunk 2]
```diff
@@ -1,49 +1,77 @@
...
+    const [background, text] = [popoverColor.text, popoverColor.background];
+
+    return css`
+      background-color: ${background};
+      color: ${text};
+    `;
+  }
+  return css`
+    background-color: ${popoverColor.background};
+    color: ${popoverColor.text};
```
Comment (ddsilva) [inline components/Popover/sub-components/Content.jsx]: Suggestion:

```js
const _getColors = ({
  skin,
  theme: {
    components: {
      popover: {
        skins: { [skin]: { text, background } },
      },
    },
  },
  inverted,
}) => `
  background-color: ${inverted ? text : background};
  color: ${inverted ? background : text};
`;
```

**Comment by allyssonsantos** [comment]
@rapahaeru 
The close icon color should change when inverted prop is true?

image

File: components/Popover/arrowProperties.js [hunk 3]
```diff
@@ -28,18 +38,18 @@ const placementPosition = {
...
     right: ${SIDE_POSITION_X}px;
-    text-shadow: 3px 1px 4px ${Colors.SHADOW[50]};
+    text-shadow: 4px 1px 4px ${Colors.SHADOW[50]};
   `,
 };
 
-const getArrow = (placement, skin) => `
-  color: ${getColorBySkin(skin)}
+const getArrow = (placement, skin, theme, inverted) => `
+  color: ${getColorBySkin(skin, theme, inverted)}
```
Comment (ggdaltoso) [inline components/Popover/arrowProperties.js]:
```suggestion
  ${getColorBySkin}
```

File: components/Popover/arrowProperties.js (hunk 1 above)
Comment (ggdaltoso) [reply components/Popover/arrowProperties.js]:
```suggestion
const getColorBySkin = ({
  skin,
  inverted,
  theme: {    
    components: {
        popover: { 
            skins: { 
                [skin]: popoverSkin
            }
        },
     },
  }
}) => `color: ${inverted ? popoverSkin.text : popoverSkin.background}`;
```

File: components/Popover/sub-components/Content.jsx [hunk 4]
```diff
@@ -1,49 +1,77 @@
...
   z-index: 100;
 
   &:before {
-    ${({ placement, skin }) => getArrow(placement, skin)};
+    ${({ placement, skin, theme, inverted }) =>
+      getArrow(placement, skin, theme, inverted)};
   }
 
-  ${({ skin }) => getStyleBySkin(skin)}
+  ${_getColors}
```
Comment (ggdaltoso) [inline components/Popover/sub-components/Content.jsx]:
```suggestion
  ${getColors}
```

File: components/Popover/sub-components/Content.jsx [hunk 5]
```diff
@@ -1,49 +1,77 @@
...
-  return `
-      background-color: ${
-        skin === 'default' ? Colors.WHITE : Colors[indexColor][200]
-      };
-      color: ${
-        skin === 'default' ? Colors.BLACK[700] : Colors[indexColor][900]
-      };
+import { shadow } from '../../shared';
+
+const _getThemeSpacing = ({ spacing: { xsmall, medium } }) => ({
```
Comment (ggdaltoso) [inline components/Popover/sub-components/Content.jsx]: Is `_getThemeSpacing` really needed?

File: components/Popover/sub-components/Content.jsx [hunk 6]
```diff
@@ -3,47 +3,77 @@ import ReactDOM from 'react-dom';
...
-        skin === 'default' ? Colors.BLACK[700] : Colors[indexColor][900]
-      };
-  `;
-};
+const _getTextColor = ({
+  skin,
+  theme: {
+    components: {
+      popover: {
+        skins: { [skin]: closeButtonColor },
```
Comment (ddsilva) [inline components/Popover/sub-components/Content.jsx]:
```suggestion
        skins: { [skin]: { closeButtonColor: { text, background } } },
```

File: components/Popover/sub-components/Content.jsx [hunk 7]
```diff
@@ -3,47 +3,77 @@ import ReactDOM from 'react-dom';
...
+      },
+    },
+  },
+  inverted,
+}) =>
+  `color: ${inverted ? closeButtonColor.background : closeButtonColor.text};`;
```
Comment (ddsilva) [inline components/Popover/sub-components/Content.jsx]:
```suggestion
  `color: ${inverted ? background : text};`;
```

**Pull request merged**